OpenRA.Mods.Common/UpdateRules/Rules/20190314/ReplaceSpecialMoveConsiderations.cs

```diff
@@ -0,0 +1,69 @@
+#region Copyright & License Information
+/*
+ * Copyright 2007-2019 The OpenRA Developers (see AUTHORS)
+ * This file is part of OpenRA, which is free software. It is made
+ * available to you under the terms of the GNU General Public License
+ * as published by the Free Software Foundation, either version 3 of
+ * the License, or (at your option) any later version. For more
+ * information, see COPYING.
+ */
+#endregion
+
+using System.Collections.Generic;
+using System.Linq;
+
+namespace OpenRA.Mods.Common.UpdateRules.Rules
+{
+	public class ReplaceSpecialMoveConsiderations : UpdateRule
+	{
+		public override string Name { get { return "Replaced special-case movement type considerations"; } }
+		public override string Description
+		{
+			get
+			{
+				return "Removed AlwaysConsiderTurnAsMove from Mobile and ConsiderVerticalMovement\n" +
+					"from GrantConditionOnMovement. Add 'Turn' and/or 'Vertical' on new ValidMovementTypes\n" +
+					"field on WithMoveAnimation and GrantConditionOnMovement instead.";
+			}
+		}
+
+		public override IEnumerable<string> UpdateActorNode(ModData modData, MiniYamlNode actorNode)
+		{
+			var mobileNode = actorNode.ChildrenMatching("Mobile").FirstOrDefault(m => !m.IsRemoval());
+			var considerTurnAsMove = false;
```

Comment:
Autodetection will fail when `Mobile` is inherited, so this won't be reliable.  Can we change this to report uses of `GrantConditionOnMovement` and `WithMoveAnimation` for the modder to check and manually update instead?